interfaces/zookeeper/v0/schema.py

```diff
@@ -0,0 +1,101 @@
+"""This file defines the schemas for the provider and requirer sides of the zookeeper_client interface.
+
+It must expose two interfaces.schema_base.DataBagSchema subclasses called:
+- ProviderSchema
+- RequirerSchema
+"""
+
+from typing import Annotated, TypeAlias
+
+from interface_tester.schema_base import DataBagSchema
+from pydantic import (
+    BaseModel,
+    Field,
+    IPvAnyAddress,
+    conint,
+)
+
+
+class Endpoint(BaseModel):
+    ip: IPvAnyAddress
+    port: conint(ge=0, le=65535) | None
+
+    def __init__(self, value: str) -> None:
+        ip, _, port = value.partition(":")
+        if not port:
+            port = None
+        super().__init__(ip=ip, port=port)
+
+
+Endpoints: TypeAlias = str
+
+
+class ZooKeeperProviderAppData(BaseModel):
+    database: str = Field(
+        description="The parent chroot zNode granted to the requirer",
+        examples=["/myappB"],
+        title="zNode",
+    )
+
+    endpoints: Endpoints = Field(
+        description="A comma-seperated list of ZooKeeper server and ports",
+        examples=["10.141.78.133:2181,10.141.78.50:2181,10.141.78.45:2181"],
+        title="ZooKeeper endpoints",
+    )
+
+    secret_user: str = Field(
+        alias="secret-user",
+        description="",
```

Comment:
**todo:** I think you want a description here. Might be useful to also add a short comma-delimited list of what ACTUALLY goes in to our `secret_user` and `secret_tls`.

Reply:
Added more info in 31afb083cafc